tests/baselines/reference/accessorWithRestParam.js

```diff
@@ -9,22 +9,12 @@ var C = /** @class */ (function () {
     function C() {
     }
     Object.defineProperty(C.prototype, "X", {
-        set: function () {
-            var v = [];
-            for (var _i = 0; _i < arguments.length; _i++) {
-                v[_i] = arguments[_i];
-            }
-        },
+        set: function () { },

```

Comment:
This looks wrong.

This test file is wrong.

`class C { set X(...v) {} }`, which this file is testing, is invalid syntax. `set` methods, on either classes or objects, only accept exactly one valid `FormalParameter`, which does _not_ allow for `FunctionRestParameter`.

Reply:
Right, thanks.

Yet TypeScript does not allow `set` to have a rest parameter anyway (i.e. the test code won't compile).

So I guess I'll revert this baseline and add it to the list above, where I ask @mhegazy about whether or not to apply the optimization in error cases.

Reply:
Oh, the transform looks correct -- the test file having a valid-looking output at all is wrong. Invalid syntax should either throw or stay as invalid syntax 😄